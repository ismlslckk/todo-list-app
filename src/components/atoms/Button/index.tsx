import classNames from 'classnames';

const Button = ({ ...props }) => {
  const className = classNames({
    [`${props.className}`]: props.className,
  });

  return (
  // eslint-disable-next-line react/button-has-type
    <button
      {...props}
      className={className}
    >
      {props.children}
    </button>
  );
};

export default Button;
