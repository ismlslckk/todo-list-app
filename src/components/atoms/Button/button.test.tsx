import { render, screen, fireEvent } from '@testing-library/react';
import reactTestRenderer from 'react-test-renderer';
import Button from '@/components/atoms/Button';

describe('button component tests', () => {
  test('expect button rendered', () => {
    render(<Button color="primary">button</Button>);
    const button = screen.getByRole('button');

    expect(button).not.toBeNull();
  });

  test('expect button click event fired', () => {
    const handleClick = jest.fn();

    render(<Button color="primary" onClick={handleClick}>button</Button>);
    const button = screen.getByRole('button');

    fireEvent.click(button);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test('expect button color rendered correctly', () => {
    const button:any = render(<Button className="primary">button</Button>);

    expect(button.container.firstChild.classList.contains('primary')).toBe(true);
  });

  it('renders correctly', () => {
    const tree = reactTestRenderer
      .create(<Button type="button">Test Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
