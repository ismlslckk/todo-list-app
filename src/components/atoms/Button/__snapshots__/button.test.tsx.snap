// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component tests renders correctly 1`] = `
<button
  className=""
  type="button"
>
  Test Button
</button>
`;
